src/chromo/decay_afterburner.py

```diff
@@ -0,0 +1,96 @@
+import chromo
+import numpy as np
+
+
+class DecayAfterburner:
+    def __init__(self, seed=None, stable_pids=None, decaying_pids=None):
+        """Initialize pythia8 and set `stable` and `decaying` particle types.
+
+        `decaying_pids` overrides `stable_pids` if the same pid is given
+        in both lists. If both are `None`, Pythia8 defaults are applied
+        """
+        config = ["ProcessLevel:all = off", "ParticleDecays:tau0Max = 1e100"]
+        pythia8 = chromo.models.Pythia8(None, seed=seed, config=config, banner=False)
+        self.pythia = pythia8._pythia
+
+        if stable_pids is not None:
+            for pid in stable_pids:
+                self.pythia.particleData.mayDecay(pid, False)
+
+        if decaying_pids is not None:
+            for pid in decaying_pids:
+                self.pythia.particleData.mayDecay(pid, True)
+
+        self.stable_pids = stable_pids
+        self.decaying_pids = decaying_pids
+
+    def __call__(self, event):
+        """Decay particles in `event` which are set as `decaying`.
+
+        Particle is `decaying` if its pdg id is passed
+        in `decaying_pids` list of constructor or it is decaing by default
+        in Pythia8. `stable_pids` of constructor is used to set stable
+        particles.
+
+        After the call `event` contains initial particles + produced particles
+        The decayed particles has event.status == 2.
+        """
+        if len(event.pid) == 0:
+            return
+
+        # Save not_final status
```

Comment:
Actually, I am not sure about this part. Currently the initial statuses are saved if they are not 1. If the value of initial statuses are not important, then they could be all set to 2. It would be faster and easier because it doesn't require np.copy and np.where in the code